packages/esm-generic-patient-widgets-app/src/obs-table-horizontal/obs-table-horizontal.component.tsx

```diff
@@ -0,0 +1,149 @@
+import {
+  Table,
+  TableCell,
+  TableContainer,
+  TableBody,
+  TableHead,
+  TableHeader,
+  TableRow,
+  InlineLoading,
+} from '@carbon/react';
+import { CardHeader, PatientChartPagination } from '@openmrs/esm-patient-common-lib';
+import { useObs } from '../resources/useObs';
+import styles from './obs-table-horizontal.scss';
+import { useTranslation } from 'react-i18next';
+import { type ConfigObjectHorizontal } from '../config-schema-obs-horizontal';
+
+interface ObsTableHorizontalProps {
+  patientUuid: string;
+}
+
+const ObsTableHorizontal: React.FC<ObsTableHorizontalProps> = ({ patientUuid }) => {
+  const { t } = useTranslation();
+  const config = useConfig<ConfigObjectHorizontal>();
+  const { data: obss, isValidating } = useObs(patientUuid, config.showEncounterType);
+  const uniqueEncounterReferences = [...new Set(obss.map((o) => o.encounter.reference))].sort();
+  let obssGroupedByEncounters = uniqueEncounterReferences.map((reference) =>
+    obss.filter((o) => o.encounter.reference === reference),
+  );
+
+  if (config.oldestFirst) {
+    obssGroupedByEncounters.sort(
+      (a, b) => new Date(a[0].effectiveDateTime).getTime() - new Date(b[0].effectiveDateTime).getTime(),
+    );
+  } else {
+    obssGroupedByEncounters.sort(
+      (a, b) => new Date(b[0].effectiveDateTime).getTime() - new Date(a[0].effectiveDateTime).getTime(),
+    );
+  }
+
+  let tableRowLabels = config.data.map(({ concept, label }) => ({
+    key: concept,
+    header: label || obss.find((o) => o.conceptUuid === concept)?.code?.text,
+  }));
+
+  if (config.showEncounterType) {
+    tableRowLabels = [{ key: 'encounter', header: t('encounterType', 'Encounter type') }, ...tableRowLabels];
+  }
+
+  const tableColumns = React.useMemo(
+    () =>
+      obssGroupedByEncounters?.map((obss, index) => {
+        const rowData = {
+          id: `${index}`,
+          date: new Date(obss[0].effectiveDateTime),
+          encounter: { value: obss[0].encounter.name },
+        };
+
+        for (const obs of obss) {
+          switch (obs.dataType) {
+            case 'Text':
+              rowData[obs.conceptUuid] = {
+                value: obs.valueString,
+              };
+              break;
+
+            case 'Number': {
+              const decimalPlaces: number | undefined = config.data.find(
+                (ele: any) => ele.concept === obs.conceptUuid,
+              )?.decimalPlaces;
+
+              let value;
+              if (obs.valueQuantity?.value % 1 !== 0) {
+                value = obs.valueQuantity?.value.toFixed(decimalPlaces);
+              } else {
+                value = obs.valueQuantity?.value;
+              }
+              rowData[obs.conceptUuid] = {
+                value: value,
+              };
+              break;
+            }
+
+            case 'Coded':
+              rowData[obs.conceptUuid] = { value: obs.valueCodeableConcept?.coding[0]?.display };
+              break;
+          }
```

Comment:
It might be worth considering if we could have a standard method for turning an obs into a displayable value or even a standard "cell" for rendering an obs.

Reply:
The reason I bring this up is I think that for numeric obs it would be nice to take into account the reference range with the same display we have in the results viewer or the vitals app.